### PR TITLE
Allows uplinks to buy more types of Martial Arts.

### PR DESCRIPTION
**Pull request by Firecage**
Firstly, this adds a new category to the Uplink, called Martial Arts.

Now, on to the main part. This adds several new types of martial arts to the uplinks. These include:

**CQC** (Well, this actually makes it no longer be Nuke Op only. Now normal traitors can buy it aswell)
**Plasma Fist** (Yes, I know it has a move which can gib people. But honestly? It is still weaker than Sleeping Carp. Sec can just taser you, or laser you, or shotgun you. Unlike Sleeping Carp which is immune to projectiles, and if gotten on discount, can be bought alongside an Esword.)
**Bo Staff**

I am making this PR, considering we can already buy one martial arts, the martial arts which is arguably the most op of them all due to providing immunity to projectiles. And personally it would be fun and interesting to have more variety, and this would also allow these MA's to be used more, instead of just sitting in a closet, dusty, and lonely.

:cl: Firecage
add: NanoTrasen emergency bulletin: We have recently learned that the Syndicate has started training their operatives in martial arts other than Sleeping Carp. We recommend that our Security forces remains vigilant of this new threat.
/:cl:

### CODE REVIEW

**Comment by kevinz000** [comment]
Cqc is op if you have a friend with carp.
Plasma fist is overpowered gib.
Bo staff is fine
Wrestling belt is you hit them once and their limb flies off.

**Comment by Firecage** [comment]
@kevinz000 . And each of these has an easy counter. Guns. Unlike Sleeping carp which can't be hit by any projectile. And I did state Plasma Fists gibbing thing. And honestly, even if Wrestling can remove limbs, it would be simpler to just buy a Double Esword or Esword + Ebow in either of those cases to kill people more easily and more cleanly. Or just buy a revolver and hack the Autolathe.

**Comment by kevinz000** [comment]
Eh depends on if they are items or innate abilities.

**Comment by Firecage** [comment]
Also, after trading TC's in several rounds for those Martial arts. They are honestly quite inferior to Sleeping Carp. I would say the only Martial Arts which is equal to it in effectiveness if Krav Maga, since its techniques are activated via action buttons instead of combos.

**Comment by Durkel** [comment]
I specifically don't give people the wrestling belt because it breaks sprites like meat spikes did. Fix it first before you make it available.

**Comment by Firecage** [comment]
@Durkel Huh, it STILL breaks sprites? Fuck, I had thought that was actually fixed.

**Comment by Iamgoofball** [comment]
I thought it was fixed too

**Comment by PKPenguin321** [comment]
Does pulling people with wrestling belt still stun them for 5 seconds? If so, fuck that

Also plasma fist is an admin only martial art that was added to demonstrate the new martial arts system, I don't think I agree with it being added

**Comment by Iamgoofball** [comment]
also please keep in mind that the wrasslin belt is also bright fucking visible, solid gold, and overlays over ALL clothing items

**Comment by Shadowlight213** [comment]
No to plasma fist. That's supposed to be admin only.

**Comment by Firecage** [comment]
Mmmm. I am actually noticing other problems with the wrestling belt aswell, along with those which Goof and Durkel mentioned. Might actually remove them from this PR for now and probably try to fix them in another one.

**Comment by Iamgoofball** [comment]
@Firecage can you list the problems

also holy fuck plasma fist is ten times more OP then wrasslin

File: code/modules/uplink/uplink_item.dm [hunk 1]
```diff
@@ -594,15 +594,6 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
...
 	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/gang)
 
-/datum/uplink_item/martial_arts/wrestling
-	name = "Wrestling Belt"
-	desc = "It is said that those who wears this belt, instantly gains an understanding of an old Earth fighting technique \
-			known simply as 'wrestling'."
-	item = /obj/item/weapon/storage/belt/champion/wrestling
-	cost = 16
-	surplus = 0
-	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/gang)
```
Comment (Iamgoofball) [inline code/modules/uplink/uplink_item.dm]: also fuck you wrassle ops is the best

Comment (Firecage) [inline code/modules/uplink/uplink_item.dm]: Mmmm, I will concede that. Pro Wrestling Ops sounds kicking.

**Comment by calzilla1** [comment]
If this gets merged, make a bundle with all of them

**Comment by ShadowDeath6** [comment]
Doesn't plasma fist gib delete items like ei nath does?

**Comment by Armhulen** [comment]
It does.

**Comment by anconfuzedrock** [comment]
if you are addin plasmeme fist make it 20 tc not 18 with 18 they can buy an emp flashlight

**Comment by Lexorion** [comment]
Plasma fist is too good at the moment,you can stunlock (multiple?) people if you cycle intents well enough. The gibbing is comparable to beheading someone with an esword, it leaves a brain behind and it takes a while to hit the combo,making it longer should/could fix the balance issue (instakill in five hits I think) and adding cooldowns to the moves could fix the stunlock issue.
I'd **love** to have more martial art stuff in the uplink!

**Comment by Firecage** [comment]
Mmmm, I guess I could attempt to add an alternative type of Plasma Martial Arts in exchange for Plasma Fist. I'le go ask in the feedback section of the forums and also link the PR there.

**Comment by Nabski89** [comment]
Just make Plasma Fist only usable by plasma men!

**Comment by Iamgoofball** [comment]
# what is wrong with the wrestling belt aside from the sprite glitch

**Comment by PKPenguin321** [comment]
its op

**Comment by Iamgoofball** [comment]
See my above points

**Comment by PKPenguin321** [comment]
they suck wrestlin belt is op

why does a regular grab do a 4 year stun that is enough to kill off of

**Comment by Iamgoofball** [comment]
The grab can be nerfed fam

**Comment by jughu** [comment]
CQC ctrl+click harn intent instant stun then pumch it so they get thrown into a wall then keep punching to stunlock

**Comment by Firecage** [comment]
So, since the thread on the forum has nothing. Anyone on here got any ideas of what martial arts to add in the place of Plasma Fist? And what its moves should be.

**Comment by Arianya** [comment]
Possibly a "greytide" martial art focused around improvised weapons? Held chairs become better block chance, toolboxes become more robust, etc etc.

Alternatively:

Art of Bushido - Comes with a sword + sheath, similar base damage levels to the Chaplain's null rod, sharp, but with "stances" based off intent, including a deflect or projectile deleting one.

Help intent would be things like cutting zipties/bolas, removing clothing a la Zorro (with a delay, for balance reasons, probably), etc.

Disarm intent would involve... disarming the opponent, aiming for the hand and the like.

Harm intent would be pure damage, possibly making the sword comparable to a esword?

Grab intent is probably best suited to be the "anti-projectile" stance.

Obviously, no effect if the blade isn't drawn/the tator tot loses his sword.

**Comment by Firecage** [comment]
Bushido does sound pretty nice. Mmmm, the sword would probably have to be a variation of the Katana then, but with a sheath for it.(Gonna have to sprite beg I guess.) It being Esword-like in harm intent might be a bit much though.

**Comment by Arianya** [comment]
You can probably cheat and just steal the existing katana sprite for when worn, and steal the saber sheath sprite for the inventory item.

**Comment by Iamgoofball** [comment]
# hey Firecage what's the issue with wrasslin belts

**Comment by PKPenguin321** [comment]
They're op

**Comment by Firecage** [comment]
Christ Goof. Will you stop spamming that?

**Comment by Iamgoofball** [comment]
I'll stop when you answer me

File: code/modules/uplink/uplink_item.dm [hunk 2]
```diff
@@ -564,26 +564,48 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
...
+/datum/uplink_item/martial_arts/cqc
 	name = "CQC Manual"
 	desc = "A manual that teaches a single user tactical Close-Quarters Combat before self-destructing."
 	item = /obj/item/weapon/cqc_manual
-	include_modes = list(/datum/game_mode/nuclear)
 	cost = 13
 	surplus = 0
+	exclude_modes = list(/datum/game_mode/gang)
+
+/datum/uplink_item/martial_arts/plasma_fist
```
Comment (Fox-McCloud) [inline code/modules/uplink/uplink_item.dm]: Let's...not. This thing is ridiculously overpowered.

Comment (Firecage) [inline code/modules/uplink/uplink_item.dm]: Read above. Already planning to replace it with a new one.

**Comment by Iamgoofball** [comment]
# answer the question

**Comment by PKPenguin321** [comment]
op

**Comment by jughu** [comment]
Heres my martial arts its called stungloves

**Comment by KorPhaeron** [comment]
Redundant bloat sorry they're all just ways to melee people unarmed with various levels of (im)balance

**Pull request closed without merging**